app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def self.update_posts_counter(user_id)
+    User.increment_counter(:post_counter, user_id)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
 end
```

Comment:
- [ ] It would be nice to use one of the Rails Active Callbacks - `after_save` to call the #update_posts_counter method every time a new post has been saved to database, then make the update_posts_counter method private. You would have to rearrange your code as follows:

e.g
```
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
  has_many :comments
  has_many :likes

  after_save :update_posts_counter

  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end

  private

  def self.update_posts_counter(user_id)
    User.increment_counter(:post_counter, user_id)
  end
end
```